### PR TITLE
net: lib: zzhc: add fixed DNS support

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
This commit adds support for China Telecom fixed DNS.

Signed-off-by: Mirko Covizzi <mirko.covizzi@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/zzhc/zzhc.c [hunk 1]
```diff
@@ -451,6 +454,47 @@ static int check_iccid(struct zzhc *ctx)
...
+		return 0;
+	}
+
+	/* Manually set DNS */
+	err = nrf_inet_pton(NRF_AF_INET, PRIMARY_DNS, &pri_dns);
+	err |= nrf_inet_pton(NRF_AF_INET, SECONDARY_DNS, &sec_dns);
+
+	if (err) {
+		return -EIO;
+	}
```
Comment (rlubos) [inline subsys/net/lib/zzhc/zzhc.c:486]: `nrf_inet_pton()` returns 1 on success (at least according to the docs), I don't think this check is correct.

File: subsys/net/lib/zzhc/zzhc.c [hunk 2]
```diff
@@ -451,6 +454,47 @@ static int check_iccid(struct zzhc *ctx)
...
+
+	err = nrf_setdnsaddr(NRF_AF_INET, &pri_dns, sizeof(pri_dns));
+	err |= nrf_setdnsaddr(NRF_AF_INET, &sec_dns, sizeof(sec_dns));
+
```
Comment (junqingzou) [inline subsys/net/lib/zzhc/zzhc.c:490]: I think the new RPC can only set one backup DNS. So the logic would be like
- check absence of network assigned DNS, and if not assigned
- set first backup DNS server and do hostname resolution, and if fails to resolve
- set second backup DNS server and do hostname resolution
- if still fails, fail ZZHC process

File: subsys/net/lib/zzhc/zzhc.c [hunk 3]
```diff
@@ -451,6 +454,80 @@ static int check_iccid(struct zzhc *ctx)
...
+		return -EIO;
+	}
+
+	/* Hostname resolution */
+	err = nrf_getaddrinfo(hostname, NULL, &hints, &info);
+	if (err) {
+		LOG_DBG("Can't resolve hostname %s. err: %d. Trying secondary backup DNS.",
+			log_strdup(hostname), err);
+	}
+
```
Comment (junqingzou) [inline subsys/net/lib/zzhc/zzhc.c:507]: Logic below this should be included in if (err)

File: subsys/net/lib/zzhc/zzhc.c [hunk 4]
```diff
@@ -481,7 +558,8 @@ static int fsm(struct zzhc *ctx)
 		}
 
 		if (check_iccid(ctx) == 0 ||
-			check_simcard_whitelist(ctx) != 0) {
+		    check_simcard_whitelist(ctx) != 0 ||
+		    check_and_set_dns(ctx) != 0) {
```
Comment (junqingzou) [inline subsys/net/lib/zzhc/zzhc.c:562]: There will be another redundant round of DNS lookup in resolve_and_connect() if backup DNS server is used. Actually I think it'd be better to implement the logic of check_and_set_dns() in resolve_and_connect().

Comment (MirkoCovizzi) [reply subsys/net/lib/zzhc/zzhc.c:564]: @junqingzou I'm aware but for now I believe it's better to do it here early and keep it localized in one function. I'll keep thinking about it, if the positives outdo the negatives I will move it around.

File: subsys/net/lib/zzhc/zzhc.c (hunk 3 above)
Comment (MirkoCovizzi) [reply subsys/net/lib/zzhc/zzhc.c:509]: @junqingzou Thanks, forgot to add the `return 0` in case of positive outcome.

**Review by rlubos — APPROVED** [review]
No more comments from my side.

File: subsys/net/lib/zzhc/zzhc.c (hunk 4 above)
Comment (junqingzou) [reply subsys/net/lib/zzhc/zzhc.c:564]: We do not have the environment to test the backup DNS anymore, which was the case in certification phase.
Please check in your code first, while I'll make further update to include missing parts from CT spec.

**Review by junqingzou — APPROVED** [review]

**Comment by rlubos** [comment]
@MirkoCovizzi Is this still DNM?

**Pull request merged**